src/MiniProfiler.Providers.StackExchange.Redis/RedisStorage.cs

```diff
@@ -0,0 +1,263 @@
+        /// <see cref="SaveAsync(MiniProfiler)"/> call. For example in a database this is likely false, whereas in
+        /// Redis and similar it's likely true (e.g. separately adding the profiler ID to a list).
+        /// </summary>
+        public bool SetUnviewedAfterSave => true;
+
+        /// <summary>
+        /// Sets a particular profiler session so it is considered "un-viewed"
+        /// </summary>
+        /// <param name="user">The user to set this profiler ID as unviewed for.</param>
+        /// <param name="id">The profiler ID to set unviewed.</param>
+        public void SetUnviewed(string user, Guid id)
+        {
+            RedisKey key = ProfilerResultUnviewedSetKeyPrefix.Append(user);
+            RedisValue value = id.ToString();
+            _database.SetAdd(key, value);
+            _database.KeyExpire(key, CacheDuration);
+        }
+
+        /// <summary>
+        /// Sets a particular profiler session to "viewed"
+        /// </summary>
+        /// <param name="user">The user to set this profiler ID as viewed for.</param>
+        /// <param name="id">The profiler ID to set viewed.</param>
+        public void SetViewed(string user, Guid id)
+        {
+            RedisKey key = ProfilerResultUnviewedSetKeyPrefix.Append(user);
+            RedisValue value = id.ToString();
+            _database.SetRemove(key, value);
+        }
+
+        /// <summary>
+        /// Returns a list of <see cref="MiniProfiler.Id"/>s that haven't been seen by <paramref name="user"/>.
+        /// </summary>
+        /// <param name="user">User identified by the current <c>MiniProfiler.Settings.UserProvider</c></param>
+        public List<Guid> GetUnviewedIds(string user)
+        {
+            RedisKey key = ProfilerResultUnviewedSetKeyPrefix.Append(user);
+            var ids = _database.SetMembers(key);
+            return ids.Select(x => Guid.Parse(x)).ToList();
+        }
+
+        /// <summary>
+        /// Asynchronously list the latest profiling results.
+        /// </summary>
+        /// <param name="maxResults">The maximum number of results to return.</param>
+        /// <param name="start">(Optional) The start of the date range to fetch.</param>
+        /// <param name="finish">(Optional) The end of the date range to fetch.</param>
+        /// <param name="orderBy">(Optional) The order to fetch profiler IDs in.</param>
+        public async Task<IEnumerable<Guid>> ListAsync(
+            int maxResults,
+            DateTime? start = null,
+            DateTime? finish = null,
+            ListResultsOrder orderBy = ListResultsOrder.Descending)
+        {
+            // TODO: start/finish not yet supported - will simply return all results
+
+            IEnumerable<RedisValue> ids;
+            switch (orderBy)
+            {
+                case ListResultsOrder.Ascending:
+                    ids = (await _database.ListRangeAsync(ProfilerResultListKey, maxResults + 1, -1).ConfigureAwait(false)).Reverse();
```

Comment:
Is this regarding `Reverse()`? I think Redis will always return list ranges in list order - and so I've tried to optimized for what I think is the usual case (i.e. descending).